distribution/lib/Standard/Table/0.2.32-SNAPSHOT/src/Data/Matching.enso

```diff
@@ -0,0 +1,155 @@
+from Standard.Base import all
+import Standard.Base.Data.Locale
+import Standard.Base.Data.Text.Regex as Regex_Module
+
+from Standard.Table.Data.Column as Column_Module import Column
+from Standard.Table.Data.Problem_Behavior as Problem_Behavior_Module import all
+
+## Strategy for matching column names.
+type Matching_Strategy
+   ## UNSTABLE
+      Exact column matching.
+
+      A column is matched if its exact name is provided.
+   type Exact (case_sensitivity : (True | Insensitive) = True)
```

Comment:
Why not create an AST:
```py
type Comparison
    type Case_Sensitive
    type Case_Insensitive
```
or, at least `case_sensitive : Boolean`

Reply:
This is to support different locales in the case insensitive case
We might also end up with a unicode folding scenario here as well (e.g. removing accents from letters)

Reply:
We've been discussing this with @wdanilo, we want this 'mixed' structure to be able to display a nice widget which is a checkbox, but when unchecked allows to display other options.

We don't use a `Boolean`, because while for now `True | False` would suffice, we plan to extend the `Insensitive` to have additional arguments in the future (like setting the Locale for character comparison) and keeping this as a separate Atom will allow us to introduce these changes in a backwards-compatible way (as we will just add them as defaulted arguments, not breaking existing code).

However I think you are right that the name of the atom should be `Case_Insensitive`, I will change that.

Reply:
Changed the name as suggested